bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeHandlerFactory.java

```diff
@@ -0,0 +1,107 @@
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+
+/**
+ * The {@link AndroidDebugBridgeHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Miguel Álvarez - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.androiddebugbridge", service = ThingHandlerFactory.class)
+public class AndroidDebugBridgeHandlerFactory extends BaseThingHandlerFactory {
+    @Nullable
+    private AndroidDebugBridgeDiscoveryService discoveryService;
+    @Nullable
+    private ServiceRegistration<?> discoveryServiceRegistration;
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES.contains(thingTypeUID);
+    }
+
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+        if (THING_TYPE_ANDROID_DEVICE.equals(thingTypeUID)) {
+            return new AndroidDebugBridgeHandler(thing, new AndroidDebugBridgeDevice());
+        }
+        return null;
+    }
+
+    private synchronized void registerDiscoveryService(AndroidDebugBridgeBindingConfiguration config) {
+        AndroidDebugBridgeDiscoveryService androidADBDiscoveryService = discoveryService;
+        if (androidADBDiscoveryService != null) {
+            androidADBDiscoveryService.updateConfig(config);
+        } else {
+            androidADBDiscoveryService = new AndroidDebugBridgeDiscoveryService();
+            androidADBDiscoveryService.updateConfig(config);
+            discoveryService = androidADBDiscoveryService;
+            discoveryServiceRegistration = bundleContext.registerService(DiscoveryService.class.getName(),
+                    androidADBDiscoveryService, null);
+        }
+    }
+
+    private void unregisterDiscoveryService() {
+        var adbDiscoveryServiceRegistration = discoveryServiceRegistration;
+        if (adbDiscoveryServiceRegistration != null) {
+            adbDiscoveryServiceRegistration.unregister();
+            discoveryServiceRegistration = null;
+        }
+    }
+
+    @Activate
+    protected void activate(ComponentContext componentContext, Map<String, Object> config) {
+        super.activate(componentContext);
+        modified(config);
+    }
```

Comment:
You could use `ThingHandlerService` which would make this code unnecessary. The documentation for this is not yet merged, but available in the PR: https://github.com/openhab/openhab-docs/pull/1262/files#diff-c4a4d8725430bc2ea046182bfc73ac51349d989c7f6a8a6fa3001a226a09ad98R932

Reply:
I was reviewing the documentation in the pr and seems like I need to create a bridge, right now I'm just passing the binding configurations to the discovery. 
Is there any other way to access to the binding configuration from the discovery service? 
Or should I create a bridge just to configure the service discovery?
Maybe I'm missing something.
Thanks in advance.

Reply:
If your discovery implements `ThingHandlerService` you need to implement the interface method `setThingHandler(ThingHandler handler)`. Then, you can retrieve the bridge's config from the handler: `handler.getThing().getConfiguration()`

Reply:
It doesn't seem to fit my case because I don't have any bridge implemented on the binding. I configure the discovery through the binding configuration.

Reply:
Now I get it. The binding configuration is stored as properties in the `AndroidDebugBridgeHandlerFactory` service. I never done this, but if you're keen, you could retrieve the config by injecting `ConfigurationAdmin` in the discovery class:

```
    @Reference
    void setConfigurationAdmin(ConfigurationAdmin admin) {
        this.admin = admin;
    }
 
    public void ranking() throws IOException {
        Configuration configOnline =
            this.admin.getConfiguration(
                "org.fipro.ds.data.online.OnlineDataService",
                null);
        Dictionary<String, Object> propsOnline = null;
        if (configOnline != null
                && configOnline.getProperties() != null) {
            propsOnline = configOnline.getProperties();
        }
```

The string argument to `admin.getConfiguration()` must be the PID of your factory: "binding.androiddebugbridge"

Then, you should be able to obtain a config parameter by calling for example `propsOnline.get("discoveryPort")`. I'm curious if this works.